jax/experimental/jax2tf/jax2tf.py

```diff
@@ -293,6 +294,7 @@ def converted_fun_flat_with_custom_gradient(*args_flat: TfVal) -> TfVal:
                     for o, _ in out_flat_raw]
     finally:
       _shape_env = {}
+      _enable_xla = _default_enable_xla
 
```

Comment:
Sorry, I'm a little confused about the logic here, but perhaps I don't properly understand the behavior you're going for. I would have assumed the implementation would have stored the original value of `_enable_xla` in a local variable at the beginning of the function, and reset it back in the finally block here, instead of setting it to the original default. But I'm guessing you intend some different behavior?

Reply:
Either way works in this case because the conversions can never be nested. I thought that this logic makes is cleared that you always restore the default. But maybe it is confusing as you point out. I have changed it to restore the previous value.